### ISSUE TITLE
copy the exact path from the new breadcrumbs

### ISSUE THREAD

**Issue by alighali85** [microsoft/vscode#58678]
It would be really great if we can copy and share the breadcrumbs generated a 
path, so we can save a huge time when we work as a team, as we can refer to the exact point of (issue or function) we need to work on next

![screen shot 2018-09-14 at 10 45 22 am](https://user-images.githubusercontent.com/15789257/45540116-e15e3a00-b80b-11e8-8ab9-9b136c9c4581.png)

**Comment by ldsands** [comment]
Sorry for the question but I figure this is as good of place as any, I've been doing a lot of web scraping lately and now I have to extract content. This feature would be very useful for that.

So now to my actual question. Does being added to the backlog mean that it is getting close(ish) to being actively worked on?

Thank you for your time and for your work on this increadible editor! 👍

**Comment by alighali85** [comment]
@ldsands usually in Agile teams, having it in the backlog means it will be scheduled to start working on it, but when they will start actively working on it still matter of ordering the priorities for the team. the only way to figure that out (form outside) if we could see the milestone they put for the next release I think.

**Comment by davetapley** [comment]
A couple other use cases:

1. Copy JSON path (maybe as xpath) https://github.com/Microsoft/vscode/issues/71453
2. Running single tests with Python's `unittest`: https://stackoverflow.com/a/26531790/21115

**Comment by clebergriff** [comment]
I upvote this issue for `Running single tests with Python's unittest` as stated by @davetapley 
Is it possible to create an extension for this matter? If so, is there any guide that could help me doing this?

**Comment by Kilozen** [comment]
Yes!  This would be extremely helpful for communications and documentation.  It's basically a "deep link" reference to anywhere in a project. 
I keep expecting to be able to right click on the breadcrumbs / path and say "Copy as Path".  Then I spend about 10 minutes clicking around, looking for every possible way to do this expected behavior before giving up and typing the 200 character path by hand (with undetected typos). 

I would think this should be easy to implement... the path is obviously already available and the breadcrumb display already responds to being clicked on, so there must be an existing handler...

**Comment by Pithikos** [comment]
Something so simple that I miss from Atom. Just being able to see the full path of the current open file and copy it.

**Comment by SaberStrat** [comment]
I can get the relative or absolute file paths from the Explorer view of course. But this requires scrolling to the ancestor folder I want at that moment. 
Being able to right click on any element of the breakcrumb and copy a relative or absolute path to that file (or even function) would be a neat time and focus safer!

**Comment by R35007** [comment]
This issue has been there for a very long period of time. Is there any update on this ticket. This is a very needed feature. There aren't  any extensions of this. It would be great if we at least get extension api support so that we can build an extension for it.

**Comment by namugach** [comment]
Is it that hard to add one of these?
`document.getElementsByClassName("tabs-breadcrumbs")[0].innerText.replaceAll("\n", " > ")`

**Comment by Pithikos** [comment]
Any updates on this so we can just copy/paste instead of writing 200 characters each time as someone mentioned?

**Comment by richex-cn** [comment]
I found a VS Code extension [copy-json-path](https://marketplace.visualstudio.com/items?itemName=nidu.copy-json-path) that can copy JSON paths.

**Comment by richex-cn** [comment]
I found a VS Code extension [copy-json-path](https://marketplace.visualstudio.com/items?itemName=nidu.copy-json-path) that can copy JSON paths.

**Comment by Trikenstein** [comment]
Please also have a look at #179531 which has more screenshots to better illustrate the feature request.

**Comment by antoniojesusA9** [comment]
You can try this:
https://marketplace.visualstudio.com/items?itemName=scott-blair.copy-breadcrumbs

**Comment by zardoy** [comment]
@antoniojesusA9 This one also has desired functionality with `Copy Current Outline Path` command: [Outline Utils](https://marketplace.visualstudio.com/items?itemName=zardoy.outline-utils) it actually copies current outline path in the document and supports custom delimeter. For example with the body of the issue it would be `OwnProps>match`

**Comment by Trikenstein** [comment]
Sorry to insist, VSCode is reaching a status of an IDE and is not a simple text editor. Copy the path of a YAML of JSON element is something basic that should be a built-in functionality. In the duplicated issue https://github.com/microsoft/vscode/issues/179531 I had filed, I provide clear screenshots showing this feature exists in IntelliJ since +10 years.

Thanks for participants who suggest to install some extension to work around the missing feature in VSCode. For now it is much simpler for me to drag-drop the YAML/JSON file into IntelliJ and copy the key from the breadcrumbs bar.

**Comment by Trikenstein** [comment]
@zardoy  A billion years ago, mother nature was facing the same issue. Every animal was asking for legs. Some sheeps asked for 2 legs on the ground 3 legs pointing towards the sky. Some horses asked for 4 legs inline, other asks for sideway legs, etc. Contrary to VSCode, mother nature satisfied all the requests. Fast forward today, the animals which remain are those who are efficient, having the most efficient legs configuration.

Here we attempt to save the VSCode maintainers a billion years of debate and wondering. The feature request here is the one that makes sense. The breadcrumb bar is visible on the screen. It is intuitive to get the path from that visual clue. Whether it's a file path or a YAML, JSON path.

**Comment by zardoy** [comment]
Interesting. Also I fully agree with you, this context menu would make it super intuitive ![m](https://user-images.githubusercontent.com/46503702/245209879-6022f3d2-a379-4cfc-b3a8-7042acc14a50.png)

**Comment by liuxhit** [comment]
> You can try this: https://marketplace.visualstudio.com/items?itemName=scott-blair.copy-breadcrumbs

Wonderful plugin!

**Comment by V1NAY8** [comment]
So, Are we getting this useful feature ? ? ?

**Comment by s-h-a-d-o-w** [comment]
Hm, I don't see any official approval that this should be implemented, so maybe posting my implementation won't be interesting but... anyway, here's what it could look like 😅: https://github.com/s-h-a-d-o-w/vscode/tree/copy-breadcrumbs-path

![Screenshot 2023-09-23 231727](https://github.com/microsoft/vscode/assets/16936908/84f709cf-564a-4bd5-9157-4b49c061ffd8)

(This was my first time working with the codebase, so things might not be ideal...)

**Comment by turbotimon** [comment]
> > You can try this: https://marketplace.visualstudio.com/items?itemName=scott-blair.copy-breadcrumbs
> 
> Wonderful plugin!

There is also a fork with slightly more features and the code is on GitHub: https://marketplace.visualstudio.com/items?itemName=MikeCunneen.cunneen-copy-breadcrumbs

Still wish and think this should be in vscode itself rather than an extension is needed for that.
